src/localstream.js

```diff
@@ -84,25 +124,46 @@ const localstream = (function() {
 		console.log('(ReachSDK::localstream::initVideo)');
 		if (! initVideoProgress) {
 			initVideoProgress=true;
+			_getAllVideoSources().then((sources)=>{
+				videoSources=sources;
+			});
```

Comment:
First, reassigning `videoSources` is not needed, it's already done within `_getAllVideoSources`.
Second, this doesn't address the fact that your synchronous code rely on an object `videoSources` defined by an asynchronous function. The stream initialization should be done within the `resolve` function of the `_getAllVideoSources` Promise:

```javascript
_getAllVideoSources().then(sources => {
	navigator.getMedia = getUserMedia;
	mLocalStreamVideo = document.createElement('VIDEO');
	mLocalStreamVideo.muted= true;
	localVideoStreams.push(mLocalStreamVideo);
	let audioSource;
	let videoSource;

	if(sources[currentVideoSource]){
		sources = sources[currentVideoSource].deviceId;
	}

	...

});
```